crates/key-server/src/server.rs

```diff
@@ -300,6 +302,20 @@ impl Server {
                     e, req_id
                 ))
             })?;
+
+        // Record the gas cost. Only do this in permissioned mode to avoid high cardinality metrics in public mode.
+        if let Some(m) = metrics {
+            if matches!(
+                self.options.server_mode,
+                ServerMode::Permissioned { client_configs: _ }
+            ) {
+                let package = vptb.pkg_id().to_hex_uncompressed();
+                m.dry_run_gas_cost_per_package
+                    .with_label_values(&[&package])
+                    .observe(dry_run_res.effects.gas_cost_summary().computation_cost as f64);
```

Comment:
maybe we should also check there are no other costs (storage, etc) and if there are, log them?